build-tools/packages/build-cli/test/commands/generate/typetests.test.ts

```diff
@@ -0,0 +1,76 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import type { TypeData } from "@fluidframework/build-tools";
+import {
+	loadTypesSourceFile,
+	typeDataFromFile,
+} from "../../../src/commands/generate/typetests.js";
+
+describe("generate:typetests", () => {
+	const logger = {
+		log: () => assert.fail(),
+		info: () => assert.fail(),
+		warning: () => assert.fail(),
+		errorLog: () => assert.fail(),
+		verbose: () => assert.fail(),
+		logHr: () => assert.fail(),
+		logIndent: () => assert.fail(),
+	};
+
+	function forCompare(data: Map<string, TypeData>): unknown[] {
+		return [...data.entries()].map(([k, v]) => ({
+			name: k,
+			import: v.name,
+			tags: [...v.tags],
+		}));
+	}
+
+	it("rollup", () => {
```

Comment:
Can you add a little more detail either here or in the test files about the difference between rollup and direct and why we need both? I think the answer is that the two test files match the kinds of types files that we generate - either directly generated by tsc or rolled up via a tool like api-extractor or something homegrown. Since we may want to type test against both kinds, we have separate test cases.

Reply:
Done.